packages/client/src/client.ts

```diff
@@ -150,8 +154,12 @@ class Client<FetcherRequestInit extends RequestInit = RequestInit> {
     return response.json() as Promise<unknown>;
   }
 
-  private getEndpoint() {
-    if (!this.config.channelId || this.config.channelId === '1') {
+  private getEndpoint(channelId?: string) {
+    if (channelId && channelId !== '1') {
+      return `https://store-${this.config.storeHash}-${channelId}.${graphqlApiDomain}/graphql`;
```

Comment:
We've backfilled all/most default channels with the channel-specific canonical URL, so we should be able to simplify this to `https://store-${this.config.storeHash}-${channelId ?? this.config.channelId}.${graphqlApiDomain}/graphql` and enforce channel ID to be set.

Reply:
Good info, wasn't aware of this.